.github/workflows/test-gator.yaml

```
@@ -49,11 +45,106 @@ jobs:
           go-version: "1.25"
           check-latest: true
 
-      - name: Download e2e dependencies
-        run: |
-          mkdir -p $GITHUB_WORKSPACE/bin
-          echo "$GITHUB_WORKSPACE/bin" >> $GITHUB_PATH
-          make e2e-dependencies KUBERNETES_VERSION=${{ matrix.KUBERNETES_VERSION }}
-
       - name: gator test
         run: make test-gator-containerized
+
+  gator_bench_test:
+    name: "Gator Bench E2E"
+    runs-on: ubuntu-22.04
+    timeout-minutes: 10
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@95d9a5deda9de15063e7595e9719c11c38c90ae2 # v2.13.2
+        with:
+          egress-policy: audit
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@1af3b93b6815bc44a9784bd300feb67ff0d1eeb3 # v6.0.0
+
+      - name: Set up Go
+        uses: actions/setup-go@4dc6199c7b1a012772edbd06daecab0f50c9053c # v6.1.0
+        with:
+          go-version: "1.25"
+          check-latest: true
+
+      - name: Build gator
+        run: make gator
+
+      - name: Test basic Rego policy benchmark
+        run: |
+          ./bin/gator bench \
+            --filename test/gator/bench/basic/ \
+            --iterations 50 \
+            --output table
+
+      - name: Test CEL policy benchmark
+        run: |
+          ./bin/gator bench \
+            --filename test/gator/bench/cel/ \
+            --iterations 50 \
+            --engine cel \
+            --output table
+
+      - name: Test dual-engine policy benchmark
+        run: |
+          ./bin/gator bench \
+            --filename test/gator/bench/both/ \
+            --iterations 50 \
+            --output table
```

```suggestion
            --output table
            --engine all
```

because the default is Rego as of now.